app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala

```diff
@@ -346,6 +347,30 @@ object TransactionQueries extends StrictLogging {
     }
   }
 
+  /*
+   * Sum outputs and group them by the given interval type
+   * LEAST and GREATEST are here to restrict to the `from` and `to` timestamp
+   */
+  def sumAddressOutputs(
+      address: Address,
+      from: TimeStamp,
+      to: TimeStamp,
+      intervalType: IntervalType
+  ): DBActionSR[(TimeStamp, Option[U256])] = {
+    val dateGroup = QueryUtil.dateGroupQuery(intervalType)
+    sql"""
+      SELECT
+        LEAST($to, GREATEST($from, (EXTRACT(EPOCH FROM ((#$dateGroup) AT TIME ZONE 'UTC')) * 1000) - 1)) as ts,
+        SUM(amount)
+      FROM outputs
+      WHERE address = $address
+      AND main_chain = true
+      AND block_timestamp >= ${ALPH.GenesisTimestamp}
```

Comment:
unfortunately no, because for your first timestamp, even if it's 20 days ago, you need to know the balance at this given timestamp and for this you need to sum all older outputs from the beginning.
If you have millions of ALPH, but everything happens 3 months ago and you only have 1 small tx 10 days ago, your queries of latest 20 days will only give you the small tx amount if we have >= $from.
Don't know if my explanation is clear?

Reply:
That's too expensive. We will remove such endpoints gradually.